compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/JvmLower.kt

```diff
@@ -274,6 +272,7 @@ private val jvmFilePhases =
         singleAbstractMethodPhase then
         assertionPhase then
         returnableBlocksPhase then
+        sharedVariablesPhase then
```

Comment:
Why this move is required?

Reply:
If a returnable block has multiple exit points, returnableBlocksPhase will create a variable and replace each return with a store to it followed by a break. If the return is inside an inline lambda, the variable might need to be wrapped into a ref for the store to work.